lib/QoreURL.cpp

```diff
@@ -65,52 +78,75 @@ struct qore_url_private {
 	 return;
       }
 
-      char* nbuf;
+      // use std::string for a temporary buffer for exception-safety
+      std::string nbuf;
 
       // find end of hostname
-      if ((p = (char* )strchr(pos, '/'))) {
+      if ((p = (char*)strchr(pos, '/'))) {
 	 // get pathname if not at EOS
 	 path = new QoreStringNode(p);
-	 printd(5, "QoreURL::parse_intern path=%s\n", path->getBuffer());
+	 printd(5, "QoreURL::parse_intern path: '%s'\n", path->getBuffer());
 	 // get copy of hostname string for localized searching and invasive parsing
-	 nbuf = (char* )malloc(sizeof(char) * (p - pos + 1));
-	 strncpy(nbuf, pos, p - pos);
-	 nbuf[p - pos] = '\0';
+         nbuf.assign(pos, p - pos);
+         //printd(5, "QoreURL::nbuf: '%s' size: %d\n", nbuf.c_str(), nbuf.size());
       }
       else
-	 nbuf = strdup(pos);
+	 nbuf.assign(pos);
 
       // see if there's a username
       // note that nbuf here has already had the path removed so we can safely do a reverse search for the '@' sign
-      if ((p = strrchr(nbuf, '@'))) {
+      if ((p = strrchr((char*)nbuf.c_str(), '@'))) {
```

Comment:
why cast to (char*) here? `strrchr` should take `const char*` acc to the man page, so `nbuf.c_str()` should be enough...